crates/holochain/src/conductor/space.rs

```diff
@@ -177,12 +179,41 @@ impl Spaces {
     }
 
     /// Check if some target is blocked.
+    #[async_recursion::async_recursion]
     pub async fn is_blocked(
         &self,
-        input: BlockTargetId,
+        target_id_0: BlockTargetId,
         timestamp: Timestamp,
-    ) -> StateQueryResult<bool> {
-        holochain_state::block::is_blocked(&self.conductor_db, input, timestamp).await
+    ) -> DatabaseResult<bool> {
+        let target_id_1 = target_id_0.clone();
+        Ok(self.conductor_db
+            .async_reader(move |txn| holochain_state::block::query_is_blocked(&txn, target_id_1, timestamp))
```

Comment:
So, here is where the read lock is acquired on the database. Can you re-arrange this so the recursion is *inside* so we don't keep re-aquiring then releasing the lock?

```rust
async fn my_recursive_logic(txn: &Transaction) { .. }
self.conductor_db.async_reader(move |txn| my_recursive_logic(txn)).await?
```

Reply:
Actually, this could even cause a deadlock, right? If a request for a write lock came in after the outer transaction is created but before the inner recursive one, it wouldn't open the inner one, and the outer one would block the write transaction from getting acquired?

Reply:
@neonphog there's no "outer transaction" in my code, `is_blocked` is recursive but the readers are awaited serially before any recursion happens

but sure we could refactor `is_blocked` with an inner recursive function that takes a read lock as an argument, but i thought it was bad to keep locks across awaits, which is what would happen here across the recursion, right?